### PR TITLE
Add author and date metadata to all proposals

### LINKED ISSUE #247: Proposals should contain author information and date
The proposals metadata should contain information about the author and also the date the proposal was accepted. 

This information seems hard to work out from the rendered proposal page:

https://ghc-proposals.readthedocs.io/en/latest/proposals/0045-pointer-rep.html

and people often refer to I, which is not timeless when decoupled from author information.

### PR DESCRIPTION
**Pull request by isovector**
This was automatically generated by using `git blame` to determine who wrote the last line in the file (the author), and then finding the date the file was created (presumably the accepted date.)
It was generated via

```bash
#!/bin/bash

for FILE in proposals/*; do

AUTHOR=$(git blame $FILE | tail -n1 | cut -f2 -d'(' | sed 's/  \+/#/g' | cut -f1 -d'#' | cut -f1,2 -d' ' | cat <( echo -n ".. author:: ") -)
DATE=$(git log --format="format:%ai" $FILE | tail -n1 | cut -f'1' -d' ' | cat <( echo -n ".. date-accepted:: ") -)

sed -i "5i$AUTHOR" $FILE
sed -i "6i$DATE" $FILE

done
```

Closes #247

### CODE REVIEW

**Comment by int-index** [comment]
It seems that the `author` field contains the full name if available, and GitHub username otherwise. If this is policy, perhaps we should document it somewhere? And, perhaps, include both the full name and the GitHub username if available?

This additional meta-info is already useful as is, thanks for gathering it. I'm just wondering if it can be made more complete.

**Comment by isovector** [comment]
@int-index, probably this shouldn't be policy. I'll just hand-edit the few github names that are present, if I can do it without doxxing anybody. Thanks for the catch!

**Comment by nomeata** [comment]
Nice approach!

Yeah, let's not make policy, and let's not dig up names that people did not already volunteer on their Github profile.

File: proposals/0032-type-data.rst [hunk 1]
```diff
@@ -2,6 +2,8 @@ Define Kinds Without Promotion
 ==============================
 
 .. proposal-number:: 32
+.. author:: Gabor Greif
```
Comment (nomeata) [inline proposals/0032-type-data.rst]: Shouldn’t this be Iavor?

Comment (isovector) [reply proposals/0032-type-data.rst]: Looks like yes. Blame is really messy on that file! I'll fix it. Nice catch!

**Pull request merged**